lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -77,15 +87,30 @@ def schedule=(schedule_hash)
         reload_schedule!
       end
 
+      def setup_schedule(schedule_hash)
+        schedule = {}
+
+        Resque.redis.evalsha(
+          startup_sha,
+          keys: [:schedules, :persisted_schedules, :schedules_changed],
+          argv: prepare_lua_schedule(schedule_hash)
+        ).each_slice(2) do |name, config|
+          schedule[name] = decode(config)
+        end
+
+        # ensure only return the successfully saved data!
+        reload_schedule!(with: schedule)
```

Comment:
Just doing `@schedule = schedule` would be simpler, since then you don't have to add a `with` keyword argument to `reload_schedule!`